src/routes/url.ts

```diff
@@ -14,13 +13,19 @@ const saveUrl: Route<{ Body: { url: string } }> = {
 			required: ['url'],
 			properties: {
 				url: { type: 'string' },
+				id: { type: 'string' },
 			},
 		},
 	},
 	async handler(request) {
 		await validateUrl(request.body.url)
 
-		const urlRequestData = { url: request.body.url, ip: request.ip } as UrlRequestData
+		let urlRequestData = { url: request.body.url, ip: request.ip } as UrlRequestData
+		// need to make sure that has admin rights
+		if ((await this.auth.isAuthorized(request)) && request.body.id) {
+			urlRequestData = { ...urlRequestData, id: request.body.id } as UrlRequestData
```

Comment:
Can you change it to throw an error instead?

If an id is supplied but the user is not Authorized, throw an unauthorized error

Reply:
will do